shared/src/ledger/vp_host_fns.rs

```diff
@@ -333,6 +334,20 @@ where
     Ok(storage.native_token.clone())
 }
 
+/// Getting the IBC event.
+pub fn get_ibc_event(
+    _gas_meter: &mut VpGasMeter,
+    write_log: &WriteLog,
+    event_type: String,
+) -> EnvResult<Option<IbcEvent>> {
+    for event in write_log.get_ibc_events() {
+        if event.event_type == event_type {
+            return Ok(Some(event.clone()));
+        }
+    }
+    Ok(None)
```

Comment:
instead of doing a linear look-up, we can eventually get the events from a map

Reply:
I tried but I found that multiple "message" events could be issued.

Reply:
hm, but currently we're short-circuiting on the first `"message"` event we find, anyway. perhaps we should return a vector of events instead?